### PR TITLE
vmbackup: support file-based configuration for -snapshot.createURL and -snapshot.DeleteURL

### PR DESCRIPTION
**Pull request by wasim-nihal**
PR includes suggested changes for https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5973

### CODE REVIEW

File: lib/flagutil/secure_url_test.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+package flagutil
```
Comment (hagen1778) [inline lib/flagutil/secure_url_test.go:1]: In the tests I don't see the case when `authToken` will be secured. For example, `-snapshot.createURL="http://localhost:8428/snapshot/create?authKey=foobar"` - will it hide `authKey` value from the logs and /flags output?

File: lib/flagutil/secure_url.go [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+	"github.com/VictoriaMetrics/VictoriaMetrics/lib/fasttime"
+)
+
+// NewSecureURL returns new `url` flag with the given name and description.
+//
+// The url value is hidden when calling SecureURL.String() for security reasons,
+// since the returned value can be put in logs.
+// Call SecureURL.Get() for obtaining the real url value.
+func NewSecureURL(name, description string) *SecureURL {
```
Comment (hagen1778) [inline lib/flagutil/secure_url.go:20]: Maybe we should go with just `URL` without `Secure` prefix. It is unlikely we'll have `URL` and `SecureURL` types.

**Review by hagen1778 — COMMENTED** [review]
Sorry if I forgot the context. Is reading URL content from file is required for you? Or can we strip this functionality?

File: lib/flagutil/secure_url_test.go (hunk 1 above)
Comment (wasim-nihal) [reply lib/flagutil/secure_url_test.go]: Yes. Basically, the change will mask the whole of the URL as `"secret"` in logs (Just like the remoteWriteURLs).
I have covered this case(authKey) in the latest commit.

File: lib/flagutil/secure_url.go (hunk 2 above)
Comment (wasim-nihal) [reply lib/flagutil/secure_url.go]: Agree. I have addressed this in the latest commit.

**Comment by wasim-nihal** [comment]
> Sorry if I forgot the context. Is reading URL content from file is required for you? Or can we strip this functionality?

Yes, there are two things here.

- support to provision create/delete URL containing credentials as file (this is k8s secret mounted to pod). [using env variables as an alternative is not allowed in our org]
- Masking of create/delete URL (that are passed as plain text to flags), containing credentials in the logs. https://github.com/VictoriaMetrics/VictoriaMetrics/issues/5973#issuecomment-2007598571

**Comment by hagen1778** [comment]
Thanks for the PR! I created an alternative implementation here https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6152
it doesn't modify lib/snapshot that much, and preserves the URL structure by masking only sensitive info.
I'll discuss both implementations with VM team on Monday.

**Comment by wasim-nihal** [comment]
Thanks @hagen1778

**Pull request closed without merging**